README.md

```diff
@@ -0,0 +1,27 @@
+# Mini Analytics [![Actions Status](https://github.com/Max-Leopold/mini-analytics/workflows/Java%20CI/badge.svg)](https://github.com/Max-Leopold/mini-analytics/actions)
+Mini Analytics is a small scale project I've done as an internship introduction project.
+
+This project can be used to search Twitter for user written Lucene queries.
+
+## Installation
+
+To install this project on your system you need [Maven](https://maven.apache.org/download.cgi), [Docker](https://www.docker.com/get-started) and a [Twitter Developer Account](https://developer.twitter.com/en/apply-for-access.html).
+
+To download the project just use `git clone https://github.com/Max-Leopold/mini-analytics.git` to clone the project onto your machine.
+
+In order to start the project you have to put your [Twitter Developer Keys](https://developer.twitter.com/en/docs/basics/authentication/guides/access-tokens.html) in the `mini-analtics/twitterpuller/src/main/resources/twitter4j.properties` file
```

Comment:
```suggestion
In order to start the project you have to put your [Twitter Developer Keys](https://developer.twitter.com/en/docs/basics/authentication/guides/access-tokens.html) in the `mini-analytics/twitterpuller/src/main/resources/twitter4j.properties` file
```

Reply:
I was wondering, now that you are adding a web scraper as well, if you can make this more configurable (in terms of what data sources to pull in). Because the whole thing could also work without twitter.

Reply:
How do you mean that? That at startup you should be able to choose the modules that should run? e.g. no twitter but reddit?

Reply:
Couldn't you just orchestrate that through docker-dompose? Something like this?

> My approach was to create a text file that has all the services in it to run (one line, space separated).
> For example, create file services.txt that contains: db web sentry python
> Then run
> docker-compose up $(< services.txt)
> 
> only services listed in the file will run.
https://github.com/docker/compose/issues/1294

Reply:
I didn't think about actual solutions so far, but yeah, I meant something configurable, not during startup. I just thought it'd be nice if having twitter api access is optional.

Reply:
Probably the `TwitterPuller` would just crash, but everything else would work fine...so technicallllyyyy it is still works without twitter keys.

Reply:
Technically everything works but the twitter puller :D

Reply:
Did you make a decision if you are going to do something about this?

Reply:
No, I didn't yet. But I think I can do that with the approach I described earlier.

Reply:
I now added the possibility to start the project without twitter keys.